### PR TITLE
Limit number of days a query can be split into.

### PR DESCRIPTION
**Pull request by tomwilkie**
Signed-off-by: Tom Wilkie <tom.wilkie@gmail.com>

### CODE REVIEW

**Comment by bboreham** [comment]
Fixes #349 ?

Why limit this to when split-by-days is on - wouldn't it be more understandable to have a simple length limit?

**Comment by tomwilkie** [comment]
> Why limit this to when split-by-days is on - wouldn't it be more understandable to have a simple length limit?

That already covered by the limits in the chunks store (see #1059) - when you don't split there is very little chance of abuse in the frontend.  This limit is here to protect the memcaches from abuse, for instance.

I do want to me this configurable on a per-user basis though.

**Comment by tomwilkie** [comment]
@bboreham I've updated this to use the same flag we use in the querier, and as a result made it configurable per user.

File: pkg/querier/frontend/query_range.go [hunk 1]
```diff
@@ -127,6 +127,11 @@ func encodeDurationMs(d int64) string {
 	return strconv.FormatFloat(float64(d)/float64(time.Second/time.Millisecond), 'f', -1, 64)
 }
 
+func msToTime(t int64) time.Time {
```
Comment (bboreham) [inline pkg/querier/frontend/query_range.go]: I used Prometheus `timestamp.Time()` for this in #1312 - we should be consistent.

File: pkg/util/validation/limits.go [hunk 2]
```diff
@@ -55,6 +56,7 @@ func (l *Limits) RegisterFlags(f *flag.FlagSet) {
 
 	f.IntVar(&l.MaxChunksPerQuery, "store.query-chunk-limit", 2e6, "Maximum number of chunks that can be fetched in a single query.")
 	f.DurationVar(&l.MaxQueryLength, "store.max-query-length", 0, "Limit to length of chunk store queries, 0 to disable.")
+	f.IntVar(&l.MaxQueryParallelism, "querier.max-query-parallelism", 14, "Maximum number of queries will be scheduled in parallel by the frontend.")
```
Comment (bboreham) [inline pkg/util/validation/limits.go]: Please mention this change in the commit description.  Ideally a separate commit.

**Comment by bboreham** [comment]
Seems OK apart from nits.

File: pkg/util/validation/limits.go (hunk 2 above)
Comment (tomwilkie) [reply pkg/util/validation/limits.go]: :+1:

File: pkg/querier/frontend/query_range.go (hunk 1 above)
Comment (tomwilkie) [reply pkg/querier/frontend/query_range.go]: :+1:

**Pull request merged**